server/app/models/stack.rb

```diff
@@ -11,6 +11,8 @@ class Stack
   has_many :stack_revisions, dependent: :destroy
   has_many :stack_deploys, dependent: :destroy
   has_many :grid_services
+  has_many :volumes
+  embeds_many :external_volumes
```

Comment:
I did try it that way but it IMO ended up even smellier. :D The `external` volume is a kind-of reference to grid level volume, so it's not really doable with a single `Volume` model. If there would be only `:volumes` relation, I don't see a clear way to define an external volume on stack level that references a grid level volume. Yes, we can add extra attribute `external: true` for the volume, but how to convey the reference to a grid level volume? Having a real relation to a volume within volume sounds really bad IMO.

I do agree that this creates bit of repetition, but couldn't come up with any cleaner alternative. I'm perfectly fine to refactor this if any better ideas? :)